benchmark/sample_genes.py

```diff
@@ -74,6 +76,72 @@ def sample_genes(ortho_groups_fp,
     return genes
 
 
+def write_genes(genes,
+                input_faa_dir,
+                output_fa_dir,
+                output_genes_fp):
+    """ Write protein sequences of selected gene families to external files
+
+    Parameters
+    ----------
+    genes : dict of dict of set of str
+        { ogid : { taxon : set(protein(s)) }
+        return value of sample_genes or filter_paralogs
+    input_faa_dir : str
+        directory of input protein sequences from the query genome and the
+        selected taxa for comparison (FASTA format, one taxon per file)
+    output_fa_dir : str
+        directory to store output protein sequences (FASTA format, one gene
+        family per file)
+    output_genes_fp : str
+        file to store a list of selected gene families and their members.
+        format: gene1<tab>taxon1|protein1,taxon2|protein2,...
+    """
+    # generate a taxon to protein to gene family map
+    #   so complicated because it is optimized for subsequent filesystem I/O
+    prots = {}
+    for ogid in genes:
+        for taxon in genes[ogid]:
+            if taxon not in prots:
+                prots[taxon] = {}
+            for prot in genes[ogid][taxon]:
+                if prot in prots[taxon]:
+                    prots[taxon][prot].add(ogid)
+                else:
+                    prots[taxon][prot] = set([ogid])
+    # match taxa with faa filenames
+    #   so complicated because OrthoFinder trims off the version number from
+    #   an NCBI-style accession (e.g., GCF_012345.1 becomes GCF_012345)
+    taxon2file = {}
+    for file in os.listdir(input_faa_dir):
+        if file.endswith('.faa'):
+            taxon = file.split('.')[0]
+            if taxon in prots:
+                taxon2file[taxon] = file
+    # read protein sequences
+    seqs = {}
+    for taxon in taxon2file:
+        for seq in io.read(os.path.join(input_faa_dir, taxon2file[taxon]),
+                           format='fasta'):
+            id = seq.metadata['id']
+            if id in prots[taxon]:
+                seqs[id] = str(seq)
```

Comment:
Read and write operations should be moved to separate modules if possible.  It'll be easier to test / maintain in the future.